hr_timesheet_limit_sale_date/models/sale_order_line.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+from odoo.osv import expression
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+    _name = 'sale.order.line'
+
+    def _timesheet_compute_delivered_quantity_domain(self):
+        domain = super()._timesheet_compute_delivered_quantity_domain()
+        order = self.mapped('order_id')
+        if len(order) == 1:
```

Comment:
This seems broken to me. 
Consider we have two different sale orders, and we update the timesheet_limit_date on both orders, then `_compute_qty_delivered` is triggered and here, the extra domain won't be appended here, and we'd have a different result by updating two sale orders at the same time than updating the same sale orders one by one...

Reply:
it's not broken, if domain triggered on many lines which belongs to different SO we skip injecting of date
it's tested in test_multiple_line_pass
this is the hard part which I tried to explain, domain is class method and it we trying to works on object

Reply:
The test `test_multiple_line_pass` doesn't bring anything actually as there are no assertions in it.
Please add timesheets on both sale orders, with entries after and before the limit date that you're going to set afterwards, and you'll see that results are different if you write on both sale order or only on one....

Reply:
I tend to agree with @grindtildeath this is broken if the sale.order.lines passed to this method belongs to different sale.order

Reply:
if different lines passed here it's not executed, but I am looking for better way

Reply:
updated @grindtildeath @TDu